apps/ui/src/components/Sidebar.tsx

```diff
@@ -1092,46 +1102,16 @@ function ExportJupyterNB() {
     );
 
     // Generate the download link on the fly
-    let element = document.createElement("a");
-    element.setAttribute(
-      "href",
-      "data:text/plain;charset=utf-8," + encodeURIComponent(fileContent)
-    );
-    element.setAttribute("download", filename);
-
-    element.style.display = "none";
-    document.body.appendChild(element);
-    element.click();
-    document.body.removeChild(element);
+    const fileName = `${
+      repoName || "Untitled"
+    }-${new Date().toISOString()}.ipynb`;
+    const dataUrl =
+      "data:text/plain;charset=utf-8," + encodeURIComponent(fileContent);
+    downloadLink(dataUrl, fileName);
     setLoading(false);
   };
 
-  return (
-    <Button
-      variant="outlined"
-      size="small"
-      color="secondary"
-      onClick={onClick}
-      disabled={false}
-    >
-      Jupyter Notebook
-    </Button>
-  );
-}
-
-function ExportSVG() {
-  // The name should contain the name of the repo, the ID of the repo, and the current date
-  const { id: repoId } = useParams();
-  const store = useContext(RepoContext);
-  if (!store) throw new Error("Missing BearContext.Provider in the tree");
-  const repoName = useStore(store, (state) => state.repoName);
-  const filename = `${repoName?.replaceAll(
-    " ",
-    "-"
-  )}-${repoId}-${new Date().toISOString()}.svg`;
-  const [loading, setLoading] = useState(false);
-
-  const onClick = () => {
+  const exportSVG = () => {
```

Comment:
I like the original approach that makes each button a separate React component.

* The original `exportSVG` component is short and easy to understand.
* Now that you embed `exportSVG` into the long `exportJupyter` component, you lose the easy-to-understand SVG logic, and also makes the Jupyter logic even harder to read.

Reply:
If we want to reuse code, we could abstract the common logic into a helper function (e.g., the `downloadLink` function), and use it in both components.

In React, we **compose** components, not **embed** components.

Do this (compose):

```html
<Parent>
  <Child1/>
  <Child2/>
</Parent>
```

Not this (embed):

```html
<Parent>
  <Child1AndChild2/>
</Parent>
```

Reply:
I think the right refactoring is to extract Jupyter export logic into a normal TS function and put it in a `utils.ts` file.